robots/flakefinder/downloader.go

```diff
@@ -81,7 +82,19 @@ func readGcsObject(ctx context.Context, client *storage.Client, bucket, object s
 	return ioutil.ReadAll(reader)
 }
 
-func FindUnitTestFiles(ctx context.Context, client *storage.Client, bucket, repo string, pr *github.PullRequest) ([]*Result, error) {
+func readGcsObjectAttrs(ctx context.Context, client *storage.Client, bucket, object string) (attrs *storage.ObjectAttrs, err error) {
+	logrus.Infof("Trying to read gcs object attrs '%s' in bucket '%s'\n", object, bucket)
+	attrs, err = client.Bucket(bucket).Object(object).Attrs(ctx)
+	if err == nil {
```

Comment:
Can you handle the happy path in the final return and the errors here (like if err != nil) ?
It adds more consistency with the rest of the code (see https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88)

Reply:
But this would yield this code

	if err != nil {
		if err == storage.ErrObjectNotExist {
			return nil, err
		}
		return nil, fmt.Errorf("Cannot read attrs from %s in bucket '%s'", object, bucket)
	}
	return

and I'm not a great friend of nested `if` statements.

Reply:
You can still treat `if err == storage.ErrObjectNotExist` first

Reply:
You're right! Thanks!